framework/__init__.py

```diff
@@ -0,0 +1,55 @@
+import logging
+import os
+import yaml
+
+from dataclasses import dataclass, field, fields
+from mergedeep import merge
+
+log = logging.getLogger(__name__)
+
+THIS_DIR = os.path.dirname(os.path.abspath(__file__))
+DEFAULT_CONFIG_PATH = os.path.join(THIS_DIR, "default_config.yaml")
+
+
+@dataclass
+class Config:
```

Comment:
I would move config object to the common-ci-utils.

Framework default related config we can keep in this repo, and allow the `Config` class to take one argument of default_config_path to be overwritten by one we pass e.g. from this repo.

We don't need to block this PR with that, but I would suggest to open issue for this to move it to `common-ci-utils` later on.

Thoughts?

Reply:
I will take it up in follow up PR